modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache

```diff
@@ -23,11 +24,15 @@ import { BASE_PATH }                                         from '../variables'
 export class {{classname}} {
     protected basePath = '{{basePath}}';
     public defaultHeaders: Headers = new Headers();
+    public configuration: Configuration = new Configuration();
 
-    constructor(protected http: Http, @Optional()@Inject(BASE_PATH) basePath: string) {
+    constructor(protected http: Http, @Optional()@Inject(BASE_PATH) basePath: string, @Optional() configuration: Configuration) {
         if (basePath) {
             this.basePath = basePath;
         }
+        if (configuration) {
```

Comment:
If we start using a configuration it would probably be better to also put the basePath in it instead of passing it sepperatly.